agent/agent_pool.go

```diff
@@ -57,12 +55,15 @@ func (r *AgentPool) Start(ctx context.Context) error {
 	defer done()
 	setStat("🏃 Spawning workers...")
 
+	idleMon := newIdleMonitor(len(r.workers))
+
 	errCh := make(chan error)
 
 	// Spawn each worker "in parallel" (in its own goroutine)
 	for _, worker := range r.workers {
 		go func() {
-			errCh <- r.runWorker(ctx, worker)
+			defer idleMon.markDead(worker)
```

Comment:
If we use `idleMon.markIdle` here, it will work as well? do we need a concept of "dead" in idle monitor?

Reply:
That makes sense.